ml-prims/src/linalg/svd.h

```diff
@@ -228,47 +218,43 @@ void svdReconstruction(math_t *U, math_t *S, math_t *V, math_t *out, int n_rows,
  * @param k: number of singular values to be computed, 1.0 for normal SVD
  * @param tol: tolerance for the evaluation
  * @param cublasH cublas handle
- * @param mgr device allocator for temporary buffers during computation
+ * @param allocator device allocator for temporary buffers during computation
  * @{
  */
 template <typename math_t>
 bool evaluateSVDByL2Norm(math_t *A_d, math_t *U, math_t *S_vec, math_t *V,
                          int n_rows, int n_cols, int k, math_t tol,
                          cublasHandle_t cublasH, cudaStream_t stream,
-                         DeviceAllocator &mgr) {
+                         std::shared_ptr<deviceAllocator> allocator) {
```

Comment:
Could pass in cumlHandle_impl instead of cuBLAS handle, cuSolver handle and allocator.

Reply:
`cumlHandle_impl` is currently part of cuML and this is ml-prims.

Reply:
Ah right. I guess we might need to discuss how we want to handle the cuML <-> ml-prims at some point.

Reply:
Doh! I forgot to file an issue for this. Let me do that now :)

Reply:
I have filed issue #498 to discuss this improvement.

Reply:
@jirikraus for the purpose of this PR, can I mark such `cumlHandle_impl` + ml-prims comments to be resolved?

Reply:
Yes that is fine by me. Thanks for filing the issue.

Reply:
Thanks Jiri. Marking this as resolved.